crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -1053,5 +1053,47 @@ static_assert(is_subtype_of(TypeOf[C.foo], object))
 static_assert(not is_subtype_of(object, TypeOf[C.foo]))
 ```
 
+### Classes with `__call__`
+
+```py
+from typing import Callable
+from knot_extensions import TypeOf, is_subtype_of, static_assert, is_assignable_to
+
+class A:
+    def __call__(self, a: int) -> int:
+        return a
+
+a = A()
+
+static_assert(is_subtype_of(A, Callable[[int], int]))
+static_assert(not is_subtype_of(A, Callable[[], int]))
+static_assert(not is_subtype_of(Callable[[int], int], A))
+
+def _(f: Callable[[int], int]) -> None: ...
+
+_(a)
+```
+
+### Bounded methods
+
+```py
+from typing import Callable
+from knot_extensions import TypeOf, static_assert, is_subtype_of
+
+class A:
+    def f(self, a: int) -> int:
+        return a
+
+    @classmethod
+    def g(cls, a: int) -> int:
+        return a
+
+a = A()
+
+static_assert(is_subtype_of(TypeOf[a.f], Callable[[int], int]))
+static_assert(is_subtype_of(TypeOf[a.g], Callable[[int], int]))
+static_assert(is_subtype_of(TypeOf[A.g], Callable[[int], int]))
```

Comment:
maybe add some negative tests in here as well, to show that we are actually checking the signature?

Would also be interesting in future to test that `A.f` is a subtype of `Callable[[A, int], int]`, but this requires self types, so not for this PR. But a test with a TODO comment would be useful here.